source/library/ServantSerf/Module.hs

```diff
@@ -0,0 +1,42 @@
+module ServantSerf.Module where
+
+import qualified Data.List as List
+import qualified Data.Maybe as Maybe
+import qualified ServantSerf.Type.Config as Config
+import qualified ServantSerf.Type.Context as Context
+import qualified ServantSerf.Type.ModuleName as ModuleName
+import qualified System.FilePath as FilePath
+
+generate :: Context.Context -> [FilePath] -> String
+generate context files =
+  let
+    source = Context.source context
+    config = Context.config context
+    apiName = Config.apiName config
+    serverName = Config.serverName config
+    moduleName = case Config.moduleName config of
+      Nothing ->
+        maybe "Main" ModuleName.toString
```

Comment:
This `"Main"` here feels unsatisfying. I'm not entirely sure what to do instead though. I could:

- Try to convert the source file name into a module name when building the context. That way if it fails I could throw an exception. 
- Change the default to an invalid module name. That would require the user to pass in `--module-name=Whatever` to solve the problem. 
- Keep `"Main"` because either it will work (hooray!) or GHC will complain that the module name doesn't match what it expected.
  - I'm not sure how GHC determines which module name it expects. It would be great if I could re-use that somehow, but I'm not sure how.